src/session/session_api.c

```diff
@@ -700,6 +700,10 @@ __session_rebalance(WT_SESSION *wt_session, const char *uri, const char *config)
 
 	SESSION_API_CALL(session, rebalance, config, cfg);
 
+	/* In-memory doesn't support rebalance operations. */
+	if (F_ISSET(S2C(session), WT_CONN_IN_MEMORY))
+		goto err;
+
```

Comment:
I don't think so. I think it's possible that at some point in the future we might need schema-worker to work on in-memory configurations, so I'm hesitant to have it enforce that level of policy.

I reviewed where schema-worker is called from:
* rebalance, ignored in `__session_rebalance`
* salvage, failed in `__session_salvage`
* upgrade, failed in `__session_upgrade` (added in 3a6dc62) 
* verify, failed in `__session_verify`
* compact, ignored in `__wt_session_compact`
* checkpoint, failed in `__session_checkpoint`
* backup cursors, failed in `__backup_start` (added in c5e9978)